File: ws/sisalema/u1/ws08CD_JavaClasses/ChickenFarmerSimulator/src/ec/edu/espe/farmsimulator/model/ChickenFarmen.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package ec.edu.espe.farmsimulator.model;

/**
 *
 *@author Yostin Sisalema,Code masters, DCCO-ESPE
 */
public class ChickenFarmen {
    private int id;
    private String name;
    private ChickenCoop[] coops;

    public ChickenFarmen(int id, String name, ChickenCoop[] coops) {
        this.id = id;
        this.name = name;
        this.coops = coops;
    }
    
    public String toString(String type) {
    StringBuilder chickenFarmer = new StringBuilder();
    String chickenData = "";
    if (coops != null) {
        for (ChickenCoop coop : coops) {
            chickenFarmer.append(coop.toString(type)).append("\n");
        }
    }
    if (type.equals("csv")) { // csv
        chickenData = "ChickenFarmen, id=" + id + ", name=" + name + ", \n" 
                    + chickenFarmer.toString();
    } else if (type.equals("txt")) { // txt
        chickenData = "ChickenFarmen: id=" + id + ", name=" + name + ": \n" 
                    + chickenFarmer.toString() + "\n";
    } else if (type.equals("json")) { // json
        chickenData = "{\n"+ "  \"ChickenFarmen\": {\n"+ "    \"id\": \"" + id + "\",\n"+ "    \"name\": \"" + name + "\",\n"+ "    \"chickenFarmer\": [\n"+ chickenFarmer.toString() + "\n"+ "    ]\n"+ "  }\n"+ "}\n";
    }
        return chickenData;
    }
    
        public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public ChickenCoop[] getCoops() {
        return coops;
    }

    public void setCoops(ChickenCoop[] coops) {
        this.coops = coops;
    }
}

File: ws/sisalema/u1/ws08CD_JavaClasses/ChickenFarmSimulatorJavascript/farm_simulator/model/ChickenFarmen.js
<<<<<<< HEAD
export class ChickenFarmen {
    constructor(id, name, coops) {
        this.id = id;
        this.name = name;
        this.coops = coops;
    }
    
    toString(type) {
        let chickenFarmer = "";
        if (this.coops != null) {
            for (let coop of this.coops) {
                chickenFarmer += coop.toString(type) + "\n";
            }
        }
        
        let chickenData = "";
        if (type === "csv") { // csv
            chickenData = "ChickenFarmen, id=" + this.id + ", name=" + this.name + ", \n" 
                        + chickenFarmer;
        } else if (type === "txt") { // txt
            chickenData = "ChickenFarmen: id=" + this.id + ", name=" + this.name + ": \n" 
                        + chickenFarmer + "\n";
        } else if (type === "json") { // json
            chickenData = "{\n"
                        + "  \"ChickenFarmen\": {\n"
                        + "    \"id\": \"" + this.id + "\",\n"
                        + "    \"name\": \"" + this.name + "\",\n"
                        + "    \"chickenFarmer\": [\n"
                        + chickenFarmer + "\n"
                        + "    ]\n"
                        + "  }\n"
                        + "}\n";
        }
        return chickenData;
    }
    
    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getCoops() {
        return this.coops;
    }

    setCoops(coops) {
        this.coops = coops;
    }
}
=======
export class ChickenFarmen {
    constructor(id, name, coops) {
        this.id = id;
        this.name = name;
        this.coops = coops;
    }
    
    toString(type) {
        let chickenFarmer = "";
        if (this.coops != null) {
            for (let coop of this.coops) {
                chickenFarmer += coop.toString(type) + "\n";
            }
        }
        
        let chickenData = "";
        if (type === "csv") { // csv
            chickenData = "ChickenFarmen, id=" + this.id + ", name=" + this.name + ", \n" 
                        + chickenFarmer;
        } else if (type === "txt") { // txt
            chickenData = "ChickenFarmen: id=" + this.id + ", name=" + this.name + ": \n" 
                        + chickenFarmer + "\n";
        } else if (type === "json") { // json
            chickenData = "{\n"
                        + "  \"ChickenFarmen\": {\n"
                        + "    \"id\": \"" + this.id + "\",\n"
                        + "    \"name\": \"" + this.name + "\",\n"
                        + "    \"chickenFarmer\": [\n"
                        + chickenFarmer + "\n"
                        + "    ]\n"
                        + "  }\n"
                        + "}\n";
        }
        return chickenData;
    }
    
    getId() {
        return this.id;
    }

    setId(id) {
        this.id = id;
    }

    getCoops() {
        return this.coops;
    }

    setCoops(coops) {
        this.coops = coops;
    }
}
>>>>>>> 8e40484f0077d7fe10786aa1aa0f3cc51debe068
